homeassistant/components/usps.py

```diff
@@ -0,0 +1,124 @@
+https://home-assistant.io/components/usps/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.dt import now
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['myusps==1.1.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'usps'
+DATA_USPS = 'data_usps'
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+COOKIE = 'usps_cookies.pickle'
+
+CONF_CAM_INTERVAL = 'interval'
+DEFAULT_CAM_INTERVAL = 10
+
+USPS_TYPE = ['sensor', 'camera']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_CAM_INTERVAL, default=DEFAULT_CAM_INTERVAL):
+            cv.positive_int
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Use config values to set up a function enabling status retrieval."""
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    name = conf.get(CONF_NAME)
+    cam_interval = conf.get(CONF_CAM_INTERVAL)
+
+    import myusps
+    try:
+        cookie = hass.config.path(COOKIE)
+        session = myusps.get_session(username, password, cookie_path=cookie)
+    except myusps.USPSError:
+        _LOGGER.exception('Could not connect to My USPS')
+        return False
+
+    hass.data[DATA_USPS] = USPSData(session, name, cam_interval)
+
+    for component in USPS_TYPE:
+        discovery.load_platform(hass, component, DOMAIN, {}, config)
+
+    return True
+
+
+class USPSData(object):
+    """Stores the data retrieved from USPS.
+
+    For each entity to use, acts as the single point responsible for fetching
+    updates from the server.
+    """
+
+    def __init__(self, session, name, cam_interval):
+        """Initialize the data oject."""
+        self._session = session
+        self._name = name or DOMAIN
+        self._cam_interval = cam_interval
+        self._packages = []
+        self._mail = []
+        self._attr = None
+        self._mail_img = []
+
+    @property
+    def name(self):
+        """Return name for sensors/camera."""
+        return self._name
+
+    @property
+    def cam_interval(self):
+        """Return interval to update mail camera images."""
+        return self._cam_interval
+
+    @property
+    def attribution(self):
+        """Return attribution for sensors/camera."""
+        return self._attr
+
+    @property
+    def packages(self):
+        """Get latest update if throttle allows. Return status."""
+        return self._packages
+
+    @property
+    def mail(self):
+        """Get latest update if throttle allows. Return status."""
+        return self._mail
+
+    @property
+    def session(self):
```

Comment:
You don't really need all these properties if you just want to return the corresponding attribute. You decide though.